akka-actor/src/main/scala/akka/dispatch/Mailbox.scala

```diff
@@ -290,7 +290,10 @@ private[akka] abstract class Mailbox(val messageQueue: MessageQueue)
      * if we closed the mailbox, we must dump the remaining system messages
      * to deadLetters (this is essential for DeathWatch)
      */
-    val dlm = actor.dispatcher.mailboxes.deadLetterMailbox
+    val dlm: Mailbox =
+      // MICRO-OPT: access of deadLetterMailbox is slow enough to avoid if not needed
+      if (messageList.nonEmpty) actor.dispatcher.mailboxes.deadLetterMailbox
```

Comment:
ok, so normally the `messageList` is always empty.
`actor` is volatile so I guess that could explain why it matters?

Reply:
Aha, that's a good explanation. I'll add a comment about that.